#! /bin/sh
# Ignore whitespace is necessary because the original context of the patch
# contains whitespace errors
patch -p0 -f --ignore-whitespace $* < $0
exit $?

2013-04-18  Andre Heinecke  <aheinecke@intevation.de>

	* pthread.h: Handle that struct timespec might be defined
    on windows without _UWIN definition.

--- pthread.h.orig      2013-04-18 15:52:11.470656381 +0000
+++ pthread.h   2013-04-18 15:52:14.762630601 +0000
@@ -110,7 +110,6 @@
 #endif

 #ifdef _UWIN
-#   define HAVE_STRUCT_TIMESPEC 1
 #   define HAVE_SIGNAL_H        1
 #   undef HAVE_CONFIG_H
 #   pragma comment(lib, "pthread")
@@ -227,6 +226,12 @@
 #include <setjmp.h>
 #include <limits.h>

+// Check if someone defined timespec
+#if defined(_UWIN) || defined(_TIMESPEC_DEFINED)
+#   define HAVE_STRUCT_TIMESPEC 1
+#endif
+
+
 /*
  * Boolean values to make us independent of system includes.
  */
